core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -601,14 +499,11 @@ public HttpResponse convertResponse(ServiceRequestContext ctx,
             }
 
             assert f != null;
-            return HttpResponse.from(f.handle((aggregated, cause) -> {
-                if (cause != null) {
-                    return handleExceptionWithContext(exceptionHandler, ctx, ctx.request(), cause);
-                }
+            return HttpResponse.from(f.thenApply(aggregated -> {
```

Comment:
Question: Just wanted to make sure that this is working even the `f` is completed exceptionally. Right?

Reply:
If `f` is completed exceptionally, the exception is propagated to the returned `HttpResponse`.
https://github.com/line/armeria/blob/6f21290a3e6e3bb2af22d8a9061d472dcabda545/core/src/main/java/com/linecorp/armeria/common/DeferredHttpResponse.java#L54-L56